### ISSUE TITLE
Support for line spacing expansion in ebook mode

### ISSUE THREAD

**Issue by hilspilsen** [sumatrapdfreader/sumatrapdf#476]
This has been my favorite pdf reader for several years. Beautiful and lightweight.
Would love to have a customisation for line spacing expansion in ebook mode, would make reading much easier.
Many thanks,
Hils

**Comment by zipzen** [comment]
+1 Line spacing for EPUB reading would make a great program almost perfect. BTW Sumatra is a blessing.

**Comment by Balur86** [comment]
+1 for Line spacing for EPUB. Just what I'm missing right now. Great app anyway.

**Comment by Phoenix-Brothres** [comment]
[FBReader](https://fbreader.org/) has line spacing option. It's also an open source project.

**Comment by Zenadix** [comment]
+1, paragraph spacing is also critical.

**Comment by sergeevabc** [comment]
Err… Hello?

**Comment by yEngineer7** [comment]
+1. Absolutely tiresome to read without ability to change line spacing.

**Comment by GitHubRulesOK** [comment]
Line Spacing is to a degree set by the author by their choice of foundry type which controls default leading (which in turn controls line height)
Here is an unmodified source ePub simply visually altered by my choice of font in advanced settings.
One is Georgia another one is a less common MS font and both may appear different in either interface.
![image](https://user-images.githubusercontent.com/12754913/102002741-8facfe80-3cf7-11eb-8e83-c56ab7eb13f8.png)

**Comment by rkmfl** [comment]
+1 for line-spacing to be added in the config/"Advanced Settings" stylesheet.

I've used just about every ePub reader (other than the Win-Store versions) on Windows, and Sumatra is the only one where I haven't been able to either customize, or expand/contract the line spacing.  It's present in literally every other reader.

Thank you :)

**Comment by GitHubRulesOK** [comment]
FBReader is **not open-source** since 2015.
FBReader supports many ebook formats using its own book parser and rendering engine. 
FBReader SDK does **not include PDF **

Linespacing cannot be adjusted in PDF
Linespacing in ePub is set by the author, you need to edit the authors settings in an editor such as calibre.

**Comment by rkmfl** [comment]
@GitHubRulesOK - 
There might be some miscommunication.  

First -- people are *only* referring to .ePub
Second -- Before claiming that ePubs should not allow line-spacing adjustments, you should download *any* major eBook reader and take a look.  I use a few eBook readers and every one of them allows line-spacing to be adjusted.

Because ePub's are typically rendered with CSS engines (of course it's just as straightforward if they're rendered in another manner and the author's spacing is observed and implemented) -- it's quite easy to either override or adjust the author's spacing based upon a multiplier.

For example, if the author specified 15 points and the eBook viewer offered a "Increase / Decrease" line-spacing option, it might adjust by 20% in each instance (eg .. 9, 12, 15, 18, 21 points).

There is absolutely nothing about the author's spacing that makes it mutually-exclusive to adjust line-spacing in the eBook reader, and as mentioned, this is very likely the reason it's offered in every popular eBook rendering tool.

Thank you.

**Comment by GitHubRulesOK** [comment]
@kjk 
MuPDF Draw - GL supposedly lets users nominate custom CSS to override author ePub settings and also allows other ePub font scaling e.g. ><. can those user preferences be assimilated/incorporated into SumatraPDF?

**Comment by kjk** [comment]
Yes, mupdf does allow custom CSS which allows, I assume, customization like arbitrary fonts, font sizes, line spacing etc.

Not yet sure how best to expose this.

**Comment by sergeevabc** [comment]
[STDU Viewer](https://en.wikipedia.org/wiki/STDU_Viewer), a great app for reading.

![image](https://user-images.githubusercontent.com/373718/205884337-e1b1304c-2208-4e66-ba80-28751307e91b.png)
